include/zephyr/bluetooth/conn.h

```diff
@@ -196,6 +196,21 @@ struct bt_conn_le_data_len_param {
 	BT_CONN_LE_DATA_LEN_PARAM(BT_GAP_DATA_LEN_MAX, \
 				  BT_GAP_DATA_TIME_MAX)
 
+/** Connection Type */
+typedef enum __packed {
```

Comment:
No reason to `typedef` this :) Suggest to keep as `enum bt_conn_type_t`

Reply:
I thought `__packed` can only be applied on `typedef`s, but if this is wrong, I can make it a simple `enum bt_conn_type` again.

Reply:
FWIW, I have nothing against a making a typedef for connection types, however I'm also not sure where and how exactly `__packed` can be used when it comes to enums. Maybe it's possible to not have it in the enum definition but only use it when declaring a variable - try it out and see what works.